src/main/java/at/hannibal2/skyhanni/data/GetFromSackData.kt

```diff
@@ -0,0 +1,130 @@
+    private val inventoryMap = mutableMapOf<Int, List<PrimitiveItemStack>>()
+
+    private var lastTimeOfCommand = SimpleTimeMark.farPast()
+
+    var sackList = emptyList<NEUInternalName>()
+        private set
+
+    /** Do not use this use [GetFromSackAPI.get] instead*/
+    fun addToQueue(items: List<PrimitiveItemStack>) = queue.addAll(items)
+
+    /** Do not use this use [GetFromSackAPI.slot] instead*/
+    fun addToInventory(items: List<PrimitiveItemStack>, slotId: Int) = inventoryMap.put(slotId, items)
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (queue.isNotEmpty() && lastTimeOfCommand.passedSince() >= minimumDelay) {
+            val item = queue.poll()
+            LorenzUtils.sendCommandToServer("gfs ${item.name.asString()} ${item.amount}")
+            lastTimeOfCommand = SimpleTimeMark.now()
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        inventoryMap.clear()
+    }
+
+    @SubscribeEvent
+    fun onSlotClicked(event: GuiContainerEvent.SlotClickEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (event.clickedButton != 1) return // right click
+        addToQueue(inventoryMap[event.slotId] ?: return)
+        inventoryMap.remove(event.slotId)
+        event.isCanceled = true
+    }
+
+    @SubscribeEvent
+    fun onTooltipRender(event: LorenzToolTipEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        val list = inventoryMap[event.slot.slotIndex] ?: return
+        event.toolTip.let { tip ->
+            tip.add("")
+            tip.add("§ePress right click to get from sack:")
+            tip.addAll(list.map { "§ex" + it.amount.toString() + " " + it.name.asString() })
+        }
+    }
+
+    fun commandHandler(args: Array<String>) {
+        if (!config.queuedGFS) {
+            LorenzUtils.sendCommandToServer("gfs ${args.joinToString(" ")}")
+            return
+        }
+
+        if (args.size != 2) {
+            LorenzUtils.chat("§cMissing arguments! Usage: /getfromsacks <name/id> <amount>", prefix = false)
+            return
+        }
+
+        val item = args[0].asInternalName()
+
+        if (!sackList.contains(item)) {
+            LorenzUtils.chat("§cCouldn't find an item with this name or identifier!", prefix = true)
+            return
+        }
+
+        val amountString = args[1]
+
+        if (!amountString.isInt()) {
+            LorenzUtils.chat("§cInvalid amount!", prefix = false)
+            return
+        }
+
+        val amount = amountString.toInt()
+
+        if (config.bazaarGFS && !LorenzUtils.noTradeMode) {
+            val sackInfo = SackAPI.fetchSackItem(item)
+            if (sackInfo.getStatus() != SackStatus.CORRECT && sackInfo.getStatus() != SackStatus.ALRIGHT) {
+                LorenzUtils.clickableChat("§fUnsure if items are available in Sack, §eCLICK§f to open bazaar", "bz ${item.asString()}")
```

Comment:
Maybe use a different color. white doesnt match SkyHannis color style and clickable messages